From 844dbbffb302c364f387b8c2a66ce820d7d4fa9d Mon Sep 17 00:00:00 2001
From: Liam Middlebrook <lmiddlebrook@nvidia.com>
Date: Tue, 29 Sep 2020 15:45:49 -0700
Subject: [PATCH] winevulkan: Add --xml argument to make_vulkan.

This eases development and of Vulkan extensions by allowing developers
to locally override the specification XML file used by winevulkan.

Signed-off-by: Liam Middlebrook <lmiddlebrook@nvidia.com>
Signed-off-by: Piers Daniell <pdaniell@nvidia.com>
Signed-off-by: Alexandre Julliard <julliard@winehq.org>
---
 dlls/winevulkan/make_vulkan | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/dlls/winevulkan/make_vulkan b/dlls/winevulkan/make_vulkan
index 2cddb161d69..23897b71856 100755
--- a/dlls/winevulkan/make_vulkan
+++ b/dlls/winevulkan/make_vulkan
@@ -3137,6 +3137,7 @@ def download_vk_xml(filename):
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument("-v", "--verbose", action="count", default=0, help="increase output verbosity")
+    parser.add_argument("-x", "--xml", default=None, type=str, help="path to specification XML file")
 
     args = parser.parse_args()
     if args.verbose == 0:
@@ -3148,8 +3149,12 @@ def main():
 
     set_working_directory()
 
-    vk_xml = "vk-{0}.xml".format(VK_XML_VERSION)
-    download_vk_xml(vk_xml)
+    if args.xml:
+        vk_xml = args.xml
+    else:
+        vk_xml = "vk-{0}.xml".format(VK_XML_VERSION)
+        download_vk_xml(vk_xml)
+
     registry = VkRegistry(vk_xml)
     generator = VkGenerator(registry)
 
